swarm.cabal

```diff
@@ -310,13 +310,18 @@ library
                       unification-fd                >= 0.11  && < 0.12,
                       unordered-containers          >= 0.2.14 && < 0.3,
                       vector                        >= 0.12 && < 0.14,
-                      vty                           >= 5.33 && < 5.39,
+                      vty                           >= 6.0 && < 7.0,
+                      vty-crossplatform             >= 0.2.0.0 && < 1.0,
                       wai                           >= 3.2 && < 3.3,
                       warp                          >= 3.2 && < 3.4,
                       witch                         >= 1.1.1.0 && < 1.3,
                       witherable                    >= 0.4 && < 0.5,
                       word-wrap                     >= 0.5 && < 0.6,
-                      yaml                          >= 0.11 && < 0.11.12.0,
+                      yaml                          >= 0.11 && < 0.11.12.0
+    if os(windows)
+      build-depends:  vty-windows                   >= 0.1.0.3 && < 1.0
+    else
+      build-depends:  vty-unix                      >= 0.1.0.0 && < 1.0
```

Comment:
Why not use only vty-crossplatform? 😕

Reply:
The only reason for the explicit dependency on unix/windows libraries is because the ColorMode setting is not part of the default configuration anymore, but down in the unix/windows settings. 

That is an unfortunate consequence of the design choice we made in how to split up the unix/windows implementations.

Reply:
So now that I'm thinking about this, I wonder if we made a mistake by putting ColorMode in the UnixSettings/WindowsSettings data types. It is something that we could potentially add to the VtyUserConfig
since both implementations support it.
That would make one less reason for users to need to have an explicit dependency on the unix/windows plugins.
@jtdaugherty - what do you think?

Reply:
Looking back at the change history, this was an oversight. I moved the color mode *detection* code to `vty-unix`, but there was a lot going on with the split-up of `Config` that the color mode setting got moved in the shuffle. Since the `ColorMode` type lives in `vty`, it makes sense to me to put that in the Vty user config as well, even if the backend packages still do their own detection. I think the caveat would be that backends might need to ignore or override the setting if it's unsuitable, but that was already true before Vty 6.

Reply:
I'll look into the implementation and see what this would involve.